modules/fedimint-mint-client/src/lib.rs

```diff
@@ -45,6 +49,138 @@ use crate::output::{
+
+    /// Subscribe to updates on the progress of a reissue operation started with
+    /// [`MintClientExt::reissue_external_notes`].
+    async fn subscribe_reissue_external_notes_updates(
+        &self,
+        operation_id: OperationId,
+    ) -> anyhow::Result<BoxStream<ReissueExternalNotesState>>;
+}
+
+/// The high-level state of a reissue operation started with
+/// [`MintClientExt::reissue_external_notes`].
+#[derive(Debug, Clone)]
+pub enum ReissueExternalNotesState {
+    /// The operation has been created and is waiting to be accepted by the
+    /// federation.
+    Created,
+    /// We are waiting for blind signatures to arrive but can already assume the
+    /// transaction to be successful.
+    Issuing,
+    /// The operation has been completed successfully.
+    Done,
+    /// Some error happened and the operation failed.
+    Failed(String),
+}
+
+#[apply(async_trait_maybe_send!)]
+impl MintClientExt for Client {
+    async fn reissue_external_notes(
+        &self,
+        notes: TieredMulti<SpendableNote>,
+    ) -> anyhow::Result<OperationId> {
+        let (mint_client_instance, mint_client) = mint_client(self);
+
+        let operation_id: OperationId = notes.consensus_hash().into_inner();
+        if self.get_operation(operation_id).await.is_some() {
+            bail!("We already reissued these notes");
+        }
+
+        let mint_input = mint_client
+            .create_input_from_notes(operation_id, notes)
+            .await?;
+
+        let tx = TransactionBuilder::new().with_input(mint_input.into_dyn(mint_client_instance));
+
+        let operation_meta_gen = |txid| MintMeta::Reissuance {
+            out_point: OutPoint { txid, out_idx: 0 },
+        };
+
+        self.finalize_and_submit_transaction(
+            operation_id,
+            MintCommonGen::KIND.as_str(),
+            operation_meta_gen,
+            tx,
+        )
+        .await
+        .expect("Transactions can only fail if the operation already exists, which we checked previously");
+
+        Ok(operation_id)
+    }
+
+    async fn subscribe_reissue_external_notes_updates(
+        &self,
+        operation_id: OperationId,
+    ) -> anyhow::Result<BoxStream<ReissueExternalNotesState>> {
+        let operation = self
+            .get_operation(operation_id)
+            .await
+            .ok_or(anyhow!("Operation not found"))?;
+
+        if operation.operation_type() != MintCommonGen::KIND.as_str() {
+            bail!("Operation is not a mint operation");
+        }
+
+        let out_point = match operation.meta() {
+            MintMeta::Reissuance { out_point } => out_point,
+        };
+
+        let (_, mint_client) = mint_client(self);
+
+        let tx_accepted_future = self.await_tx_accepted(operation_id, out_point.txid);
+        let output_finalized_future = mint_client.await_output_finalized(operation_id, out_point);
+
+        Ok(Box::pin(stream! {
+            yield ReissueExternalNotesState::Created;
```

Comment:
WTH?! :D

Reply:
`yield` via macro ^^ I actually saw it first in an @okjodom PR